rust/private/clippy.bzl

```diff
@@ -225,6 +225,7 @@ rust_clippy_aspect = aspect(
         ),
     },
     provides = [ClippyInfo],
+    required_providers = [rust_common.crate_info],
```

Comment:
`rust_shared_library` and `rust_static_library` do not emit a `CrateInfo` directly (because their  use case is to produce artifacts that are then used from other languages), but we still want to be able to run clippy and rustfmt on them. This should work:
```
required_providers = [[rust_common.crate_info], [rust_common.test_crate_info]] 
```

Reply:
That's a good question, and looks like we currently don't run the checks on targets that don't provide `CrateInfo` already. I'll look into reenabling them independently of this PR.

Reply:
Got clippy and rustfmt working with rules that provide `TestCrateInfo`. I updated the tests for `rustfmt` but not `clippy`. Maybe someone could add that in a separate PR?

Reply:
Sounds good!